src/auth/auth.service.ts

```diff
@@ -21,89 +21,124 @@ export class AuthService {
     this.logger.log(`Validating user: ${username}`);
-    const user = await this.userService.getByUsername(username);
+    try {
+      const user = await this.userService.getByUsername(username);
 
-    if (!user) {
-      this.logger.warn(`User not found: ${username}`);
-      return null;
-    }
+      if (!user) {
+        this.logger.warn(`User not found: ${username}`);
+        return null;
+      }
 
-    if (!user.password) {
-      this.logger.error(`Password is missing for user: ${username}`);
-      throw new Error('Password is missing for the user');
-    }
+      if (!user.password) {
+        this.logger.error(`Password is missing for user: ${username}`);
+        throw new Error('Password is missing for the user');
+      }
 
-    const isPasswordValid = await bcrypt.compare(password, user.password);
-    if (!isPasswordValid) {
-      this.logger.warn(`Invalid password for user: ${username}`);
-      return null;
-    }
+      const isPasswordValid = await bcrypt.compare(password, user.password);
+      if (!isPasswordValid) {
+        this.logger.warn(`Invalid password for user: ${username}`);
+        return null;
+      }
 
-    this.logger.log(`User validated successfully: ${username}`);
-    const { password: _, ...result } = user;
-    return result;
+      this.logger.log(`User validated successfully: ${username}`);
+      const { password: _, ...result } = user;
+      return result;
+    } catch (error) {
+      this.logger.error(
+        `Error occurred while validating user: ${username}. Error: ${error.message}`,
+      );
+      throw new Error('Validation failed');
+    }
   }
 
   async login(user: any) {
-    this.logger.log(`User ${user.username} is logging in`);
-    const payload: JwtPayload = {
-      username: user.username,
-      id: user.id,
-      roleId: user.roleId,
-    };
-    const token = this.jwtService.sign(payload);
-    this.logger.log(`JWT token generated for user: ${user.username}`);
-    return { access_token: token };
+    this.logger.log(`User ${user.username} is attempting to log in`);
+    try {
+      const payload: JwtPayload = {
+        username: user.username,
+        id: user.id,
+        roleId: user.roleId,
+      };
+      const token = this.jwtService.sign(payload);
+      this.logger.log(`JWT token generated for user: ${user.username}`);
+      return { access_token: token };
+    } catch (error) {
+      this.logger.error(
+        `Failed to log in user: ${user.username}. Error: ${error.message}`,
+      );
+      throw new Error('Login failed');
+    }
   }
 
   generateJwtToken(user: UserDto): string {
-    if (!user || !user.id || !user.username) {
-      this.logger.error('Invalid user data for JWT generation');
-      throw new Error('Cannot generate token: Invalid user data');
+    try {
+      if (!user || !user.id || !user.username) {
+        this.logger.error('Invalid user data provided for JWT generation');
```

Comment:
This is not neccesary, because in next line you throwing error, that also logs. So it will cause duplicate logs